app/src/main/java/ch/epfl/sdp/contamination/ConcreteCachingDataSender.java

```diff
@@ -42,7 +42,10 @@ void setInteractor(GridFirestoreInteractor interactor) {
     public CompletableFuture<Void> registerLocation(Carrier carrier, Location location, Date time) {
         refreshLastPositions(time, location);
         Map<String, Object> element = new HashMap<>();
-        element.put("geoPoint", new GeoPoint(location.getLatitude(), location.getLongitude()));
+        element.put("geoPoint", new GeoPoint(
+                location.getLatitude()/CachingDataSender.EXPAND_FACTOR,
```

Comment:
can't you add a test to see if the getLatitude is not already in [-99,99 ] and do the division only if it is not ? 
In that way it is more safe

Reply:
I don't think it's a good idea to consider that our schema could change on a whim. It's unlikely that the schema will change in the foreseeable future

Reply:
ok

Reply:
but I don't know if every part of our program will use this datasender correctly

Reply:
Did you talk with Matteo about that ?
I will accept your PR but trust you you will discuss that point before

Reply:
You should use the constant COORDINATE_PRECISION from GridFirestoreInteractor

Reply:
Actually, I think there is a bigger problem here. The conversion from int (as appearing in the grid label on Firestore) to a valid double should happen **before** creating the Location (argument of registerLocation) @TheCrazyKing